### ISSUE TITLE
[Git Flow] Finish hotfix branch: Make tagging optional

### ISSUE THREAD

**Issue by jnehlmeier** [ForkIssues/Tracker#1650]
When finishing a hotfix branch Fork automatically tags the commit using the name of the hotfix branch. Because the build process creates a new version tag for this merged hotfix anyways, I would love to see an option in Fork to disable automatic tag creation when finishing a hotfix. Or does this option already exist but I haven't found it?

Also because Fork creates a tag the merge commit message for the development branch states `Merge tag 'xyz' into development`. Without tagging this should change to `Merge branch 'hotfix/xyz' into development`

**Comment by DanPristupov** [comment]
GitFlow is deprecated. So, truth be told, we don't take much attention to it.

**Comment by jnehlmeier** [comment]
I don't think it is deprecated because it is a workflow. Every workflow has it pro/cons and it has to support your release practice. With CI/CD and micro services sure it does not fit well. However if you release quarterly or every half year then it still works perfectly fine, assuming you don't have many micro services. Every project is different, right?

If you think about it, trunk based development isn't that different from git-flow anyways. Trunk based development has release branches to support bug fixing via cherry picking. These release branches are basically the same as the master branch of git-flow. Feature branches are used locally in both models. 
The only real difference is that git-flow starts hotfixes from master/main branch while trunk based development picks them from master/main branch (which represents is the development branch in git-flow). So the direction of creating hotfixes is different. And that trunk based development uses cherry-picking is a consequence of using dedicated release branches.

Anyways, git tagging should be controllable by developers. But if you are not interested in git-flow anymore I stick with sourcetree app for now, as it allows toggling tag creation on/off.

**Comment by DanPristupov** [comment]
> I don't think it is deprecated because it is a workflow. Every workflow has it pro/cons and it has to support your release practice.

Yes, but why do you need a library for that? Just create feature/release/hotfix branches and merge them manually. It's very easy.

**Comment by jnehlmeier** [comment]
> > I don't think it is deprecated because it is a workflow. Every workflow has it pro/cons and it has to support your release practice.
> 
> Yes, but why do you need a library for that? Just create feature/release/hotfix branches and merge them manually. It's very easy.

Yes, sure. But with way more repetitive mouse clicks:

- switch to main
- open context menu on hotfix branch
- select merge
- hit merge button again
- switch to dev branch
- open context menu on hotfix branch
- select merge
- hit merge button again
- open context menu on hotfix branch
- select delete
- hit delete button again

versus:

- Open branch menu in toolbar
- select `finish hotfix`
- hit `finish hotfix` button
